ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/CreateTableDesc.java

```diff
@@ -937,7 +936,8 @@ public Table toTable(HiveConf conf) throws HiveException {
             tbl.getTTable().getDictionary() : new ObjectDictionary();
         List<ByteBuffer> buffers = new ArrayList<>();
         String statsSetup = StatsSetupConst.ColumnStatsSetup.getStatsSetupAsString(true,
-            storageHandler != null && storageHandler.isMetadataTableSupported() ? "metadata" : null, // Skip metadata directory for Iceberg table
+            // Is it an Iceberg table? ignore all leftover files
+            storageHandler != null && storageHandler.isMetadataTableSupported(),
```

Comment:
nit: Method `storageHandler.storageHandler()` is introduced by HIVE-25457 to identify if a table has metada table, it is not really used for check iceberg table.

Reply:
Thank you for the comment. The `storageHandler.isMetadataTableSupported()` is only enabled in Iceberg table currently, 
checking with the storage handler APIs, there is no such api to tell whether the table is Iceberg based or not.
I'm also thinking of checking whether the `storageHandler ` belongs to Iceberg storage handler, but that is too Iceberg specified.

Reply:
https://github.com/apache/hive/blob/e2b36e4449506b632e4fb68cd4f48f9a572407ee/ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLUtils.java#L224-L225
yes, `storageHandler.isMetadataTableSupported()` is only used for icebeg now. But in current Hive code, we often use snippet like this to check iceberg table.

Reply:
This property is added in meta hook, I'm afraid we cannot get this before creating the table.

Reply:
Thanks for letting me know this that parameter` table_type` does not exist when creating iceberg table. I found this parameter exists at the time of table creating: `storage_handler -> org.apache.iceberg.mr.hive.HiveIcebergStorageHandler`.  Maybe we can use this parameter to check iceberg table when creating table?  
Just for your reference,

Reply:
btw, can't we rely on storageHandler.canProvideColStatistics()?

Reply:
@zhangbutao, Thank you for the suggestion. I did something similar in the original PR, it looked too Iceberg specific.
 
Use `canProvideColStatistics` instead as @deniskuzZ mentioned.